stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -358,7 +357,7 @@ internal constructor(
                 logger.logE("[setUser] Failed to connect user. Please check you don't have connected user already.")
                 Result.error(ChatError("Failed to connect user. Please check you don't have connected user already."))
             }
-        }.onError { disconnect() }
```

Comment:
As far as I remember this change was introduced not so long ago to avoid having weird states when the connection fails. Are you sure we should remove it? Maybe instead we change the logic when setting the same user once they are already set? @JcMinarro  what do you think?

Reply:
We shouldn't try to `connectUser` if the user is already connected. Let's solve this together with @JcMinarro

Reply:
@amitkma it is not to be disconnected when the app go to background. 
It is for the case when the connection `connectUser()` is not completed sucessfully (Network error, token error, timeout...)
On that case we need to keep `ChatClient` disconnected